vscripts/gamemodes/custom_aimtrainer/_aimtrainer.nut

```diff
@@ -2007,14 +2007,25 @@ void function ChallengeWatcherThread(float endtime, entity player)
 			Signal(player, "ChallengeTimeOver")
 		}
 	)
-	
 	while(true){
 		if(!AimTrainer_INFINITE_CHALLENGE) 
 			if(Time() > endtime) break
 		WaitFrame()
 	}
 }
 
+// Passes the call back to OnStraferDummyDamaged only if the damage is from grenade explosion
```

Comment:
Would this be able to just be implemented into the function normally? Since all other challenges don't utilise a null check, you could check if the dummy is a grenade dummy, and if damage info is null.

Reply:
Much cleaner. Updated with latest push